src/main/scala/com.snowplowanalytics.sauna/responders/urbanairship/UAResponder.scala

```diff
@@ -0,0 +1,103 @@
+package com.snowplowanalytics.sauna
+package responders
+package urbanairship
+
+import scala.io.Source.fromInputStream
+
+import java.io.InputStream
+
+import apis.urbanAirship
+
+import akka.actor.{ ActorRef, Props }
+
+import responders.Responder.FileAppeared
+
+  /**
+ * Does stuff for UrbanAirship Static List upload feature.
+ *
+ * @see https://github.com/snowplow/sauna/wiki/Urban-Airship-Responder-user-guide
+ * @param urbanairship Instance of UrbanAirship.
+ * @param logger A logger actor.
+ */
+
+class UAResponder(urbanairship: urbanAirship)(implicit logger: ActorRef) extends Responder {
+
+  import UAResponder._
+
+  val pathPattern =
+    """.*com\.urbanairship/
+      |static_lists/
+      |v1/
+      |tsv:\*/
+      |.+$
+    """.stripMargin
+      .replaceAll("[\n ]", "")
+
+  def process(fileAppeared: FileAppeared): Unit = {
+    import fileAppeared._
+    convertTSV(is)
+  }
+  
+/**
+   * Converts the TSV data and groups it in a map so as to make batched upload requests to UrbanAirship
+   *
+   * @param is the InputStream of the TSV file 
+   * @return a map of the data grouped by application keys and then by listNames
+   */
+
+  def convertTSV(is: InputStream): Unit = {
```

Comment:
First of all - it's type. It is `Unit` again. 
Which make it untestable, unpredictable, hard to read, side-effecting et cetera, et cetera. If for example we would like to check how it handles different corner cases (like incorrect TSV structure with 3 columns or escaped double-quote in the middle of column) - we couldn't write a unit test for it because we have no return value, just plain old `Unit` all the time, for all inputs.